octree_web_viewer/client/octree_viewer.ts

```diff
@@ -47,329 +47,332 @@ void main() {
+                nodeRenderData.min.z + nodeRenderData.edgeLength
+            )
+        );
+        geometry.boundingSphere = new THREE.Sphere(
+            new THREE.Vector3(
+                nodeRenderData.min.x + nodeRenderData.edgeLength / 2,
+                nodeRenderData.min.y + nodeRenderData.edgeLength / 2,
+                nodeRenderData.min.z + nodeRenderData.edgeLength / 2
+            ),
+            nodeRenderData.edgeLength / 2 * SQR3
+        );
+
+        let material = commonMaterial.clone();
+        material.uniforms = {
+            min: { value: nodeRenderData.min },
+            edgeLength: { value: nodeRenderData.edgeLength },
+            size: commonMaterial.uniforms['size'],
+            alpha: commonMaterial.uniforms['alpha'],
+            gamma: commonMaterial.uniforms['gamma'],
+        };
+        this.threePoints = new THREE.Points(geometry, material);
+        scene.add(this.threePoints);
+    }
 }
 
 export class OctreeViewer {
-  // TODO(hrapp): These are only public, so we can wire up DAT to affect
-  // material.size. If DAT supports callbacks, we can encapsulate this nicer.
-  public material: THREE.ShaderMaterial;
-  public maxLevelToDisplay: number;
-
-  private loadedData: { [key: string]: NodeData } = {};
-  private nodeLoader: NodeLoader;
-  private batches: NodeData[][] = [];
-  private currentlyLoading: number;
-  private useTransparency: boolean;
-
-  constructor(private scene: THREE.Scene, private onNewNodeData: () => void) {
-    this.material = new THREE.ShaderMaterial({
-      uniforms: {
-        size: { value: 2 },
-        alpha: { value: 1 },
-        gamma: { value: 1 },
-      },
-      vertexShader: VERTEX_SHADER,
-      fragmentShader: FRAGMENT_SHADER,
-    });
-    this.useTransparency = false;
-    this.maxLevelToDisplay = 3;
-
-    this.nodeLoader = new NodeLoader();
-    this.currentlyLoading = 0;
-  }
-
-  public alphaChanged() {
-    let newUseTransparency = this.material.uniforms['alpha'].value < 1;
-    if (newUseTransparency != this.useTransparency) {
-      this.material.transparent = newUseTransparency;
-      this.scene.traverse(function(node) {
-        if (node instanceof THREE.Points && node.material instanceof THREE.ShaderMaterial) {
-          node.material.transparent = newUseTransparency;
-        }
-      });
+    // TODO(hrapp): These are only public, so we can wire up DAT to affect
+    // material.size. If DAT supports callbacks, we can encapsulate this nicer.
+    public material: THREE.ShaderMaterial;
+    public maxLevelToDisplay: number;
+    private octreeId: string;
+
+    private loadedData: { [key: string]: NodeData } = {};
+    private nodeLoader: NodeLoader;
+    private batches: NodeData[][] = [];
+    private currentlyLoading: number;
+    private useTransparency: boolean;
+
+
+    constructor(private scene: THREE.Scene, private onNewNodeData: () => void, octreeId: string) {
+        this.material = new THREE.ShaderMaterial({
```

Comment:
Nit: you could write `private octreeId: string` in the constructor and get rid of the line `private octreeId: string;` above and the line `this.octreeId = octreeId;` below.

Reply:
done